src/lib/handlePageRequest.ts

```diff
@@ -40,32 +36,52 @@ export const handlePageRequest = (
   destinationIfValidCallback: DestinationIfValidCallback = async () =>
     destinationIfValid
 ): GetServerSideProps =>
-  withCookieRedirect(async (context: GetServerSidePropsContext) => {
-    const authGateway = new BasicAuthGateway();
-    const authUseCase = new AuthenticateUser(authGateway);
-    await authUseCase.execute(context);
-
-    const beaconsContext: BeaconsContext =
-      await decorateGetServerSidePropsContext(context);
-    const userDidSubmitForm = beaconsContext.req.method === "POST";
-
-    if (userDidSubmitForm) {
-      return handlePostRequest(
-        beaconsContext,
-        formManagerFactory,
-        transformCallback,
-        destinationIfValidCallback
+  withCookieRedirect(
+    withContainer(async (context: BeaconsGetServerSidePropsContext) => {
+      const {
+        getCachedRegistration,
+        saveCachedRegistration,
+        authenticateUser,
+      } = context.container;
+
+      await authenticateUser(context);
+
+      const beaconsContext: BeaconsContext =
+        await decorateGetServerSidePropsContext(context);
+
+      const registration: Registration = await getCachedRegistration(
+        retrieveUserFormSubmissionId(context)
       );
-    }
 
-    return handleGetRequest(beaconsContext, formManagerFactory);
-  });
+      const useIndexDoesNotExist =
+        beaconsContext.useIndex >
+        registration.getRegistration().uses.length - 1;
+      if (useIndexDoesNotExist) {
+        registration.createUse();
+        await saveCachedRegistration(beaconsContext.submissionId, registration);
+      }
```

Comment:
Needed to add this to keep the `additional-beacon-use` page clean.  It's a temporary fix here pending the full slaying of `handlePageRequest` 🐉

I took the opportunity of creating the `saveCachedRegistration` use case and using the AppContainer here as well, which I hope will make the bit-by-bit refactor easier.

Reply:
As discussed, looks okay but will have to consider users' navigating to URLs that are /register-a-beacon?useIndex=1000 which would have an unintended side effect of creating an extra use.

Have a look at how the delete API endpoint route goes and then consider encapsulating this logic to create a use in a separate API route